backend/src/main/java/org/sefglobal/core/academix/service/ItemService.java

```diff
@@ -56,22 +52,25 @@ public Item getItemById(long id) throws ResourceNotFoundException {
     /**
      * Create a new {@link Item}
      *
-     * @param subCategoryIds which are the parent {@link SubCategory}s of {@link Item}
      * @param item           which holds the data to be added
      * @return the created {@link Item}
      *
      * @throws ResourceNotFoundException is thrown if the requesting {@link Item} doesn't exist
      */
-    public Item addItem(List<Long> subCategoryIds, Item item) throws ResourceNotFoundException {
+    public Item addItem(Item item) throws ResourceNotFoundException {
+        List<Long> subCategoryIds = new ArrayList<>();
+        List<SubCategory> subCategories = item.getSubCategories();
+        for (SubCategory subCategory: subCategories){
+            subCategoryIds.add(subCategory.getId());
+        }
         List<SubCategory> existingSubCategories = subCategoryRepository.findAllById(subCategoryIds);
         if (subCategoryIds.size() > existingSubCategories.size()) {
```

Comment:
You can remove the loop
```suggestion
        List<SubCategory> subCategories = item.getSubCategories();
        List<SubCategory> existingSubCategories = subCategoryRepository.findAllById(subCategoryIds);
        if (subCategories.size() > existingSubCategories.size()) {
```

Reply:
But to get the list of subcategories, we need the list of subcategory Ids. 
`List<Long> subCategories`

Then the IDs taken from the request body are added to the List
Followed by the SubCategories relevant to each id are loaded into the `subCategories` List